### ISSUE TITLE
Add summary for build process in TravisCI

### ISSUE THREAD

**Issue by jougs** [nest/nest-simulator#99]
Currently, the log files of Travis contain different types of information in one large block of text:
* configuration log and summary
* build output
* output from the coding guideline checker
* static code analysis results
* logs from the testsuite
In case of failures, it is hard to find the exact cause of failure.

Once the log files are available on S3 (#98), the logs shown by TravisCI could be shortened to only contain summaries of the information detailed above and a link to the full logs. They could also point out more prominently, what the exact cause for the failure was.

@lekshmideepu @tammoippen: Could you please look into this?

**Comment by lekshmideepu** [comment]
As we have merged #98, we could decide about the files which are to be uploaded to S3 to avoid the confusion in the primary log output. I think, the output from the coding guideline checker and the static code analysis results can be uploaded to S3.

**Comment by heplesser** [comment]
Ideally, the build log on Travis should be compact and point to the source of problems. All "volume output" should be as artifacts on S3. Essentially, I would like to see on Travis only 

```
Setup ... passed
NEST coding guidelines check ... passed
NEST configure ... passed
NEST static code analysis ... passed
NEST build ... passed
NEST install ... passed
NEST testsuite ... passed
```

If any of those tests fails, instead of `passed`, a short message should be provided. For setup, configure, build and install, probably a tail of the console output is most suitable, for the coding guidelines/static analysis the error message and pertaining file, and for the testsuite a list of failing tests (if that is tricky, the count of failing tests as displayed on the console now). All else should then go onto S3.

Note that I put the coding guideline and static analysis early in the suggestion above, so that fast-running tests will fail early, reducing the time required for to flag failure. I suppose the static analysis must come after configure so that all  `HAVE_*` macros are defined.

**Comment by tammoippen** [comment]
Please see PR #119.

Regarding the short TravisCI log: all Travis does comes into the log, i.e. all the setup stuff will be in there anyway. If I am not mistaken, this behaviour is hard to change.

I have also put the static analysis to the front, but Travis will not fail immediately when the formatting is wrong, but will first try to build NEST and perform `make installcheck`. This way we will have more information instead of only formatting is off.

**Comment by tammoippen** [comment]
Fixed by #119
